### DISCUSSION TITLE
How to request thirdparty to supbase when protected by authelia

### DISCUSSION THREAD

**Discussion by Rameshkumardas** [authelia/authelia#7483] [Q&A] [answered]
### How to request thirdparty to https://xyz.doamin.com by api client or sdk

```python

import os
from supabase import create_client, Client

SUPABASE_URL="https://xyz.doamin.com"
SUPABASE_KEY="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"

supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)
response = supabase.table("books").select("*").execute()
print("response", response.data)

```

I have installed and protected xyz.doamin.com by authelia. Now how can i request by api and connect with fluterworkflow

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

To request a third-party API protected by Authelia, you need to ensure that your API client or SDK can handle the authentication flow required by Authelia. Here’s a step-by-step guide to achieve this:

1. **Obtain an Access Token**: First, you need to obtain an access token from Authelia using the client credentials. This involves making a POST request to the token endpoint.

2. **Use the Access Token**: Once you have the access token, you can use it to authenticate your requests to the protected API.

Here’s an example of how you can do this in Python:

```python
import requests
from supabase import create_client, Client

# Authelia OIDC Configuration
AUTH_URL = "https://auth.example.com/api/oidc/token"
CLIENT_ID = "your_client_id"
CLIENT_SECRET = "your_client_secret"

# Supabase Configuration
SUPABASE_URL = "https://xyz.domain.com"
SUPABASE_KEY = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"

# Step 1: Obtain Access Token
token_response = requests.post(
    AUTH_URL,
    data={
        'grant_type': 'client_credentials',
        'client_id': CLIENT_ID,
        'client_secret': CLIENT_SECRET
    }
)

token_response.raise_for_status()
access_token = token_response.json()['access_token']

# Step 2: Use Access Token to Authenticate Requests
headers = {
    'Authorization': f'Bearer {access_token}',
    'apikey': SUPABASE_KEY
}

supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)
response = supabase.table("books").select("*").execute(headers=headers)
print("response", response.data)
```

### Flutter Integration

For Flutter, you can use the `http` package to make authenticated requests. Here’s an example:

```dart
import 'package:http/http.dart' as http;
import 'dart:convert';

Future<void> fetchBooks() async {
  final authUrl = 'https://auth.example.com/api/oidc/token';
  final clientId = 'your_client_id';
  final clientSecret = 'your_client_secret';
  final supabaseUrl = 'https://xyz.domain.com';
  final supabaseKey = 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx';

  // Step 1: Obtain Access Token
  final tokenResponse = await http.post(
    Uri.parse(authUrl),
    body: {
      'grant_type': 'client_credentials',
      'client_id': clientId,
      'client_secret': clientSecret,
    },
  );

  if (tokenResponse.statusCode == 200) {
    final accessToken = json.decode(tokenResponse.body)['access_token'];

    // Step 2: Use Access Token to Authenticate Requests
    final headers = {
      'Authorization': 'Bearer $accessToken',
      'apikey': supabaseKey,
    };

    final response = await http.get(
      Uri.parse('$supabaseUrl/rest/v1/books'),
      headers: headers,
    );

    if (response.statusCode == 200) {
      print('Response: ${response.body}');
    } else {
      print('Failed to fetch books: ${response.statusCode}');
    }
  } else {
    print('Failed to obtain access token: ${tokenResponse.statusCode}');
  }
}
```

This example demonstrates how to obtain an access token from Authelia and use it to authenticate requests to a protected API in both Python and Flutter.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment, marked as answer]
There are multiple methods available. Simplest of which is using the Authorization Basic Scheme with the `Authorization` or `Proxy-Authorization` header depending on your configuration. Alternatively there is a flow using OAuth 2.0 described here: https://www.authelia.com/integration/openid-connect/oauth-2.0-bearer-token-usage/
